gettsim/taxes/zu_verst_eink/zu_verst_eink.py

```diff
@@ -0,0 +1,154 @@
+"""
+Calculate taxable income (zve = zu versteuerndes Einkommen). The calculation
+of the 7 branches of income is according to
+https://de.wikipedia.org/wiki/Einkommensteuer_(Deutschland)#Rechenschema
+
+The income types 1 to 3 according to the law are subsumed under the first income typ
+(business income). The distinction is important as there are different deduction rules
+for each income type. In fact, you need several taxable incomes because of
+
+- child allowance vs. child benefit
+- abgeltungssteuer vs. taxing capital income in the tariff ( not implemented yet, #81)
+
+It's always the most favorable for the taxpayer, but you know that only after
+applying the tax schedule.
+"""
+import pandas as pd
+
+
+def _zu_verst_eink_kein_kinderfreib(
+    _zu_verst_eink_kein_kinderfreib_vorläufig, kind, anz_erwachsene_in_tu, tu_id
+):
+    """
+
+    Parameters
+    ----------
+    _zu_verst_eink_kein_kinderfreib_vorläufig
+    kind
+    anz_erwachsene_in_tu
+    tu_id
+
+    Returns
+    -------
+
+    """
+
+    zve_tu = (
+        (_zu_verst_eink_kein_kinderfreib_vorläufig.loc[~kind])
+        .groupby(tu_id)
+        .transform(sum)
+    )
+    out = _zu_verst_eink_kein_kinderfreib_vorläufig * 0
+    out.loc[~kind] = zve_tu / anz_erwachsene_in_tu.loc[~kind]
+    return out
+
+
+def _zu_verst_eink_kein_kinderfreib_vorläufig(
+    sum_brutto_eink,
+    vorsorge,
+    sonderausgaben,
+    behinderungsgrad_pauschalbetrag,
+    hh_freib,
+    altersfreib,
+):
+    """
+
+    Parameters
+    ----------
+    sum_brutto_eink
+    vorsorge
+    sonderausgaben
+    behinderungsgrad_pauschalbetrag
+    hh_freib
+    altersfreib
+
+    Returns
+    -------
+
+    """
+    out = (
+        sum_brutto_eink
+        - vorsorge
+        - sonderausgaben
+        - behinderungsgrad_pauschalbetrag
+        - hh_freib
+        - altersfreib
+    ).clip(lower=0)
+    return out
+
+
+def _zu_verst_eink_kinderfreib(
+    _zu_verst_eink_kein_kinderfreib_vorläufig,
+    kind,
+    anz_erwachsene_in_tu,
+    kinderfreib,
+    tu_id,
+):
+    """
+
+    Parameters
+    ----------
+    _zu_verst_eink_kein_kinderfreib_vorläufig
+    kind
+    anz_erwachsene_in_tu
+    kinderfreib
+    tu_id
+
+    Returns
+    -------
+
+    """
+
+    zu_vers_eink_kinderfreib = (
+        _zu_verst_eink_kein_kinderfreib_vorläufig[~kind] - kinderfreib.loc[~kind]
+    )
+    zu_verst_eink_tu = (
+        (zu_vers_eink_kinderfreib.loc[~kind]).groupby(tu_id).transform(sum)
+    )
+    out = _zu_verst_eink_kein_kinderfreib_vorläufig * 0
+    out.loc[~kind] = zu_verst_eink_tu / anz_erwachsene_in_tu.loc[~kind]
+    return out
+
+
+def _ertragsanteil(jahr_renteneintr):
+    """
+    Calculate the share of pensions subject to income taxation.
+
+    Parameters
+    ----------
+    jahr_renteneintr
+
+    Returns
+    -------
+
+    """
+    out = pd.Series(index=jahr_renteneintr.index, name="_ertragsanteil", dtype=float)
+    out.loc[jahr_renteneintr <= 2004] = 0.27
```

Comment:
Are the values parameters? Should they be added params? no between.

Reply:
Added piecewise linear function to params!